### PR TITLE
refactor: suggestions for improvements to the cli code done

### PR DESCRIPTION
**Pull request by victordantasdev**

### CODE REVIEW

File: cli/bin/cli.js [hunk 1]
```diff
@@ -4,18 +4,21 @@ const open = require('open');
...
 async function main() {
+  const validFlags = ['-v', '--version', '-h', ' --help'];
   const message = process.argv[2];
 
-  if (message === undefined || message === '-h' || message === '--help') {
+  if (
+    message === undefined ||
+    message.toLowerCase() === '-h' ||
+    message.toLowerCase() === '--help'
+  ) {
```
Comment (noghartt) [inline cli/bin/cli.js:14]: If you'll use `message` just being "lower-cased", I think that you can do it:

```
const message = process.argv[2]?.toLowerCase();
```

**Review by noghartt — APPROVED** [review]

File: cli/bin/cli.js [hunk 2]
```diff
@@ -4,18 +4,21 @@ const open = require('open');
 const { showHelp, showVersion, showErrorMsg } = require('../index');
 
 async function main() {
+  const validFlags = ['-v', '--version', '-h', ' --help'];
   const message = process.argv[2];
```
Comment (daniloab) [inline cli/bin/cli.js:8]: `process.argv[2]?.toLowerCase()`

File: cli/bin/cli.js [hunk 3]
```diff
@@ -4,18 +4,21 @@ const open = require('open');
...
   } else if (
-    (message.charAt(0) === '-' && message !== '-h') ||
-    (message.charAt(0) === '-' && message !== '--help') ||
-    (message.charAt(0) === '-' && message !== '-v') ||
-    (message.charAt(0) === '-' && message !== '--version')
+    message.toLowerCase() === '-v' ||
+    message.toLowerCase() === '--version'
   ) {
+    showVersion();
+  } else if (message.charAt(0) === '-' && !validFlags.includes(message)) {
```
Comment (daniloab) [inline cli/bin/cli.js:21]: what about early return?

File: cli/bin/cli.js (hunk 1 above)
Comment (victordantasdev) [reply cli/bin/cli.js:14]: Yeah, I thought of doing it this way, but it makes the message the user wants to tweet also lowercase.
Here is a test to `node cli.js "How can I learn Relay?"`

image

This `.toLowerCase()` is just to call the help or version functions even if the user type a valid flag in uppercase.

Comment (noghartt) [reply cli/bin/cli.js:14]: Oh, I see! So, I think that you can run something like it, based on the @daniloab's comment that he mentions the *early return*:

```js
if (message.charAt(0) === '-' && !validFlags.includes(message)) {
  // ...
}

switch (message.toLowerCase()) {
  case '-v':
  case '--version':
    showVersion();
    break;
  // Others cases...
}
```

What do you think about it?

File: cli/bin/cli.js (hunk 3 above)
Comment (ftonato) [reply cli/bin/cli.js:21]: This line of code will create a problem **not allowing people** to start a twitter message with a character being the "-" (minus)

**Comment by ftonato** [comment]
Other than my comment above (**which might be a bug**), all code has been manually tested by me and everything seems to work as expected.

----

Below are screenshots of usage:

**version** command:
version-command

**help** command:
image

**invalid parameter**:
wrong-parameter

**correct parameter**:
correct-parameter

correct-parameter-browser


//cc @sibelius

**Review by ftonato — CHANGES_REQUESTED** [review]
Please take a look at the [following comment](https://github.com/sibelius/ccsseraphini/pull/40/files#r786322149).

File: cli/bin/cli.js [hunk 4]
```diff
@@ -1,30 +1,39 @@
...
-  ) {
-    showVersion();
-  } else if (message.charAt(0) === '-' && !validFlags.includes(message)) {
-    showErrorMsg(message);
-  } else {
-    return await open(
-      `https://twitter.com/intent/tweet?text=${message}%0Acc%20%40sseraphini`,
+  if (message.length > 264) {
+    return console.log(
+      color('The maximum number of characters is 264!', 'red'),
```
Comment (ftonato) [inline cli/bin/cli.js:12]: Currently we **don't have this limitation in the web version**, we let users write whatever they want, when they get to twitter they can "break the tweet into several"

**Comment by sibelius** [comment]
can you rebase ?

**Review by sibelius — COMMENTED** [review]
what about yargs?

**Comment by victordantasdev** [comment]
Do you want me to add yargs to the project? I'm trying to use as few dependencies as possible, but I have no problem adding this one.

**Comment by victordantasdev** [comment]
> what about yargs?

Do you want me to add yargs to the project? I'm trying to use as few dependencies as possible, but I have no problem adding this one.

**Comment by ftonato** [comment]
@sibelius @victordantasdev 

> what about yargs?

https://twitter.com/addyosmani/status/314785735171518464?t=LTn0i0DPP8AD4JQqlOxxWg&s=19

**Comment by sibelius** [comment]
let's do it

**Comment by victordantasdev** [comment]
I just updated the CLI to use `yargs`, should I do a new PR?

**Pull request merged**